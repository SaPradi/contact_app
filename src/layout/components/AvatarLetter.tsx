import '../styles/avatarLetter.css';
import { extractLetterAvatar } from '../../utils/string';


interface AvatarLetterProps {
    first_name: string;
    last_name: string;
}

const AvatarLetter: React.FC<AvatarLetterProps> = ({first_name,last_name}) => {

    return (
        <div className="avatar">
            <p className="avatar__letter"> {extractLetterAvatar(first_name,last_name)} </p>
        </div>
    )
}

export default AvatarLetter
