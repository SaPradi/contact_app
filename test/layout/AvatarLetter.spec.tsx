import React from "react";
import { render,screen } from "@testing-library/react";
import AvatarLetter from '../../src/layout/components/AvatarLetter';


describe('Testing in <AvatarLetter>',()=>{

    const first_name = 'Santiago';
    const last_name = 'Pradilla';

    test('Should match with the snapshot',()=>{
        const { container } = render(<AvatarLetter first_name={first_name} last_name={last_name}/>)
        expect( container ).toMatchSnapshot();
    })


    test('Should show first letter name and lastname',()=>{
        const lettersName = 'SP';
        render(<AvatarLetter first_name={first_name} last_name={last_name}/>)
        const name = screen.getByText(lettersName);
        expect( name ).toBeTruthy();
    })


})