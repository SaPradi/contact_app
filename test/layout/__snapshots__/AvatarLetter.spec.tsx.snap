// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <AvatarLetter> Should match with the snapshot 1`] = `
<div>
  <div
    class="avatar"
  >
    <p
      class="avatar__letter"
    >
       
      SP
       
    </p>
  </div>
</div>
`;
